python/cuml/test/dask/test_logistic_regression.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2019-2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+from cuml.dask.common import utils as dask_utils
+from sklearn.metrics import accuracy_score
+from sklearn.datasets import make_classification
+from sklearn.linear_model import LogisticRegression as skLR
+import pandas as pd
+import numpy as np
+import dask_cudf
+import cudf
+
+pytestmark = pytest.mark.mg
+
+
+def _prep_training_data(c, X_train, y_train, partitions_per_worker):
+    workers = c.has_what().keys()
+    n_partitions = partitions_per_worker * len(workers)
+    X_cudf = cudf.DataFrame.from_pandas(pd.DataFrame(X_train))
+    X_train_df = dask_cudf.from_cudf(X_cudf, npartitions=n_partitions)
+
+    y_cudf = np.array(pd.DataFrame(y_train).values)
+    y_cudf = y_cudf[:, 0]
+    y_cudf = cudf.Series(y_cudf)
+    y_train_df = \
+        dask_cudf.from_cudf(y_cudf, npartitions=n_partitions)
+
+    X_train_df, \
+        y_train_df = dask_utils.persist_across_workers(c,
+                                                       [X_train_df,
+                                                        y_train_df],
+                                                       workers=workers)
+    return X_train_df, y_train_df
+
+
+def make_classification_dataset(datatype, nrows, ncols, n_info):
+    X, y = make_classification(n_samples=nrows, n_features=ncols,
+                               n_informative=n_info, random_state=0)
+    X = X.astype(datatype)
+    y = y.astype(datatype)
+
+    return X, y
+
+
+@pytest.mark.mg
+@pytest.mark.parametrize("nrows", [1e5])
+@pytest.mark.parametrize("ncols", [20])
+@pytest.mark.parametrize("n_parts", [2, 23])
+@pytest.mark.parametrize("fit_intercept", [False, True])
+@pytest.mark.parametrize('datatype', [np.float32, np.float64])
+def test_lr(nrows, ncols, n_parts, fit_intercept,
```

Comment:
Since this class is using a different code path for input type processing, it might be worth it to also add dask_cudf and dask_cupy tests. Probably tying one datatype be a dask_cudf and the other dask_cupy to avoid duplicating the test matrix unnecessarily